tests/pipeline/test_quarters_estimates.py

```diff
@@ -0,0 +1,898 @@
+                            q1e1,
+                            q1e2,
+                            q2e1,
+                            q2e2,
+                            sid):
+        return pd.DataFrame({
+            EVENT_DATE_FIELD_NAME: cls.q1_release_dates + cls.q2_release_dates,
+            'estimate': [.1, .2, .3, .4],
+            FISCAL_QUARTER_FIELD_NAME: [1.0, 1.0, 2.0, 2.0],
+            FISCAL_YEAR_FIELD_NAME: [2015.0, 2015.0, 2015.0, 2015.0],
+            TS_FIELD_NAME: [q1e1, q1e2, q2e1, q2e2],
+            SID_FIELD_NAME: sid,
+        })
+
+    @classmethod
+    def init_class_fixtures(cls):
+        # Must be generated before call to super since super uses `events`.
+        cls.events = cls.gen_estimates()
+        super(WithEstimatesT0, cls).init_class_fixtures()
+
+    def get_expected_estimate(self,
+                              q1_knowledge,
+                              q2_knowledge,
+                              comparable_date):
+        return pd.DataFrame()
+
+    def test_estimates(self):
+        dataset = QuartersEstimates(1)
+        engine = SimplePipelineEngine(
+            lambda x: self.loader,
+            self.trading_days,
+            self.asset_finder,
+        )
+        results = engine.run_pipeline(
+            Pipeline({c.name: c.latest for c in dataset.columns}),
+            start_date=self.trading_days[0],
+            end_date=self.trading_days[-1],
+        )
+        for sid in self.sids:
+            sid_estimates = results.xs(sid, level=1)
+            ts_sorted_estimates = self.events[
+                self.events[SID_FIELD_NAME] == sid
+            ].sort(TS_FIELD_NAME)
+            for i, date in enumerate(sid_estimates.index):
+                comparable_date = date.tz_localize(None)
+                # Filter out estimates we don't know about yet.
+                ts_eligible_estimates = ts_sorted_estimates[
+                    ts_sorted_estimates[TS_FIELD_NAME] <= comparable_date
+                ]
+                # If there are estimates we know about:
+                if not ts_eligible_estimates.empty:
+                    # Determine the last piece of information we know about
+                    # for q1 and q2. This takes advantage of the fact that we
+                    # only have 2 quarters in the test data.
+                    q1_knowledge = ts_eligible_estimates[
+                        ts_eligible_estimates[FISCAL_QUARTER_FIELD_NAME] == 1
+                    ]
+                    q2_knowledge = ts_eligible_estimates[
+                        ts_eligible_estimates[FISCAL_QUARTER_FIELD_NAME] == 2
+                    ]
+                    expected_estimate = self.get_expected_estimate(
+                        q1_knowledge,
+                        q2_knowledge,
+                        comparable_date,
+                    )
+                    if not expected_estimate.empty:
+                        for colname in sid_estimates.columns:
```

Comment:
There are extra columns there because it's the raw data, which includes sid and timestamp.